File: dataforge-core/src/main/java/hep/dataforge/io/text/SimpleMarkupRenderer.java
package hep.dataforge.io.text;

import java.io.OutputStream;
import java.io.PrintStream;

/**
 * A simple renderer using basic PrintStream without color support
 * Created by darksnake on 03-Jan-17.
 */
public class SimpleMarkupRenderer extends StreamMarkupRenderer {
    private final PrintStream stream;

    public SimpleMarkupRenderer() {
        this.stream = System.out;
    }

    public SimpleMarkupRenderer(PrintStream stream) {
        this.stream = stream;
    }

    public SimpleMarkupRenderer(OutputStream stream) {
        this.stream = new PrintStream(stream);
    }

    @Override
    protected void printText(String string) {
        stream.print(string);
    }

    @Override
    protected void ln() {
        stream.println();
        stream.flush();
    }
}

File: dataforge-core/src/main/kotlin/hep/dataforge/io/markup/SimpleMarkupRenderer.kt
package hep.dataforge.io.markup

import java.io.OutputStream
import java.io.PrintStream

/**
 * A simple renderer using basic PrintStream without color support
 * Created by darksnake on 03-Jan-17.
 */
class SimpleMarkupRenderer(private val stream: PrintStream = System.out) : StreamMarkupRenderer() {

    constructor(stream: OutputStream) : this(PrintStream(stream))

    override fun printText(string: String) {
        stream.print(string)
    }

    override fun ln() {
        stream.println()
        stream.flush()
    }
}
